### PR TITLE
Add execname support

### PR DESCRIPTION
**Pull request by daemon1024**
**Purpose of PR?**:

Based on discussion at https://kubearmor.slack.com/archives/C01F9V3SEHY/p1709480346589829

**Does this PR introduce a breaking change?**
NA

**If the changes in this PR are manually verified, list down the scenarios covered:**:
Added testcase

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: deployments/CRD/KubeArmorPolicy.yaml [hunk 1]
```diff
@@ -358,8 +361,6 @@ spec:
                           items:
                             type: string
                           type: array
-                      required:
-                      - path
```
Comment (daemon1024) [inline deployments/CRD/KubeArmorPolicy.yaml:362]: Ideally we want either path/bin to be required. 
CRDs do tend to support it through `oneOf` concept but kubebuilder has no API to create that validation

File: KubeArmor/core/unorchestratedUpdates.go [hunk 2]
```diff
@@ -677,21 +677,20 @@ func (dm *KubeArmorDaemon) restoreKubeArmorPolicies() {
 					}
 
 				} else { // HostSecurityPolicy
-					var hostPolicy tp.HostSecurityPolicy
+					var hostPolicy tp.K8sKubeArmorHostPolicy
```
Comment (nyrahul) [inline KubeArmor/core/unorchestratedUpdates.go:680]: Is there a difference between k8s based host policy and VM based host policy?

File: KubeArmor/BPF/enforcer.bpf.c [hunk 3]
```diff
@@ -364,12 +377,11 @@ static inline int match_net_rules(int type, int protocol, u32 eventID) {
     }
   }
 
-  bpf_map_update_elem(&bufk, &one, z, BPF_ANY);  
-  p->path[0] = dnet ;
+  bpf_map_update_elem(&bufk, &one, z, BPF_ANY);
```
Comment (nyrahul) [inline KubeArmor/BPF/enforcer.bpf.c:380]: Can we add pre-commit hooks or a github action to fail if such whitespaces are added? We can have a GFS for this.

**Review by rksharma95 — DISMISSED** [review]

**Review by nyrahul — DISMISSED** [review]
LGTM. Few minor comments.

File: KubeArmor/core/unorchestratedUpdates.go (hunk 2 above)
Comment (daemon1024) [reply KubeArmor/core/unorchestratedUpdates.go:680]: So tp.K8sKubeArmorHostPolicy = Status Field + tp.HostSecurityPolicy

No difference in actual Policy Structure

We don't use the Status Field for VM.
We send gRPC events as K8s events as well so as they are handled in a streamlined manner and they don't need special handling.

File: KubeArmor/BPF/enforcer.bpf.c (hunk 3 above)
Comment (daemon1024) [reply KubeArmor/BPF/enforcer.bpf.c:380]: These were added by clang format. I will create an issue to have clang formatting on and verify it like we do for go-fmt.

File: KubeArmor/enforcer/appArmorProfile.go
Comment (DelusionalOptimist) [inline KubeArmor/enforcer/appArmorProfile.go:1]: Need to add handling for `execname` rules in AppArmor profiles for host.

**Review by DelusionalOptimist — DISMISSED** [review]
LGTM.

**Review by DelusionalOptimist — DISMISSED** [review]

**Review by PrimalPimmy — DISMISSED** [review]

**Review by DelusionalOptimist — APPROVED** [review]

**Review by PrimalPimmy — APPROVED** [review]

**Pull request merged**